### PR TITLE
add AsyncResult.bindError

### PR DESCRIPTION
**Pull request by stoft**
### Problem

On rare occasions we need to do processing on the error track, e.g. run a compensation action.

### Solution

Add a `bindError` convenience function.

### CODE REVIEW

File: src/Insurello.AsyncExtra/AsyncExtra.fs [hunk 1]
```diff
@@ -44,10 +44,21 @@ module AsyncResult =
 
                 | Error err -> Async.singleton (Error err))
 
+    let bindError: ('errT -> AsyncResult<'x, 'errU>) -> AsyncResult<'x, 'errT> -> AsyncResult<'x, 'errU> =
+        fun f asyncResultErrT ->
+            asyncResultErrT
```
Comment (axelo) [inline src/Insurello.AsyncExtra/AsyncExtra.fs:49]: As it is the last param you don't have to name it and can just return `Async.bind`

File: src/Insurello.AsyncExtra/AsyncExtra.fs [hunk 2]
```diff
@@ -44,10 +44,21 @@ module AsyncResult =
 
                 | Error err -> Async.singleton (Error err))
 
+    let bindError: ('errT -> AsyncResult<'x, 'errU>) -> AsyncResult<'x, 'errT> -> AsyncResult<'x, 'errU> =
+        fun f asyncResultErrT ->
```
Comment (axelo) [inline src/Insurello.AsyncExtra/AsyncExtra.fs:48]: f could be named onError, or errorMapper maybe?

File: src/Insurello.AsyncExtra/AsyncExtra.fs [hunk 3]
```diff
@@ -44,10 +44,21 @@ module AsyncResult =
...
+            asyncResultErrT
+            |> Async.bind (function
+                | Ok x -> Async.singleton (Ok x)
+
```
Comment (axelo) [inline src/Insurello.AsyncExtra/AsyncExtra.fs:52]: I don't think the new line is needed for readability :)

File: src/Insurello.AsyncExtra/Insurello.AsyncExtra.fsproj [hunk 4]
```diff
@@ -12,7 +12,12 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Update="FSharp.Core" Version="4.7.2" />
     <Compile Include="AsyncExtra.fs" />
   </ItemGroup>
+  <PropertyGroup>
+    <OtherFlags>$(OtherFlags) --warnon:1182 --warnaserror:FS0025</OtherFlags>
+    <!-- <TreatWarningsAsErrors>true</TreatWarningsAsErrors> -->
```
Comment (axelo) [inline src/Insurello.AsyncExtra/Insurello.AsyncExtra.fsproj:20]: Do we want this uncommented? If removed, should enforce this while building.

in the workflow use ` /WarnAsError` for the build command or `-warnaserror` in the publish command.

Comment (stoft) [reply src/Insurello.AsyncExtra/Insurello.AsyncExtra.fsproj:20]: In the publish to nuget?

Comment (axelo) [reply src/Insurello.AsyncExtra/Insurello.AsyncExtra.fsproj]: Yeah publish was is needed, build should be enough!

**Pull request merged**